align_system/algorithms/lib/persona/provider.py

```diff
@@ -32,11 +42,82 @@
 logger = logging.getLogger(__name__)
 
 
+def _default_probe_filter(probe):
+    return True
+
+
+class ScalarAlignmentDistanceFunc:
+    def __init__(self, alignment_target_dict):
+        self.probe_values = {
+            KDMA_TO_PROBE_MAPPING[k['kdma']]: k['value'] * 10
+            for k in alignment_target_dict.get('kdma_values', [])
+            if k['kdma'] in KDMA_TO_PROBE_MAPPING
+        }
+
+    def __call__(self, backstories):
+        # Now that we have the probe values, we can find a set of
+        # backstories that maximize the value.  For each backstory,
+        # add the value
+        backstories_with_values = []
+        for backstory in backstories:
+            value = sum(
+                np.abs(probe['response_value'] - self.probe_values[probe['probe']])
+                for probe in backstory['probes']
+                if probe['probe'] in self.probe_values
```

Comment:
@DavidMChan just a heads up, I made this set of changes from the original as it was defaulting to a probe value of 0 if it wasn't in the alignment target (usually we only had the single KDMA of interest in there) so we were often just getting personas with the lowest probe values across the board (I think this was the bug we noticed in the call where the target was "Moral Judgement" of 0.0 and one of the three backstories had a value of "5").
```
-                np.abs(probe['response_value'] - self.probe_values.get(probe['probe'], 0))
+                np.abs(probe['response_value'] - self.probe_values[probe['probe']])
                 for probe in backstory['probes']

```